platform/admin/core/admin-core-configpolicy/src/main/java/org/codice/ddf/admin/config/policy/AdminConfigPolicy.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.config.policy;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.shiro.authz.Permission;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+import ddf.security.permission.CollectionPermission;
+import ddf.security.permission.KeyValueCollectionPermission;
+import ddf.security.permission.KeyValuePermission;
+import ddf.security.policy.extension.PolicyExtension;
+
+public class AdminConfigPolicy implements PolicyExtension {
+
+    public static final String FEATURE_NAME = "feature.name";
+
+    public static final String SERVICE_PID = "service.pid";
+
+    public static final String VIEW_FEATURE_ACTION = "view-" + FEATURE_NAME;
+
+    public static final String VIEW_SERVICE_ACTION = "view-" + SERVICE_PID;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdminConfigPolicy.class);
+
+    Map<String, List<KeyValueCollectionPermission>> featurePolicyPermissions = new HashMap<>();
+
+    Map<String, List<KeyValueCollectionPermission>> servicePolicyPermissions = new HashMap<>();
+
+    @Override
+    public KeyValueCollectionPermission isPermittedMatchAll(
+            CollectionPermission subjectAllCollection,
+            KeyValueCollectionPermission matchAllCollection) {
+        return isPermittedMatchOne(subjectAllCollection, matchAllCollection);
+    }
+
+    @Override
+    public KeyValueCollectionPermission isPermittedMatchOne(
+            CollectionPermission subjectAllCollection,
+            KeyValueCollectionPermission matchOneCollection) {
+
+        if (matchOneCollection.getAction() == null || (!matchOneCollection.getAction()
+                .equals(VIEW_FEATURE_ACTION) && !matchOneCollection.getAction()
+                .equals(VIEW_SERVICE_ACTION))) {
+            return matchOneCollection;
+        }
+
+        KeyValueCollectionPermission newMatchCollection = new KeyValueCollectionPermission();
+
+        for (Permission permission : matchOneCollection.getPermissionList()) {
+
+            if (!(permission instanceof KeyValuePermission)) {
+                continue;
```

Comment:
My thinking behind this policy extension is that any feature or service not configured to have a permission requirement is white listed. At this point we know the permission is requesting to view a feature or service by checking the action. If the permission is of an unknown class or the name of the service or feature is unknown to the policy extension then it is white listed and should be permitted. Only if the policy extension is configured to have permissions for the given feature or service should the requirements be matched with the subject collection to authorize, at which point we should only add the permission to the returning collection if the subject did not have the required credentials. Probably would be a good idea to put this in a comment as class description… Hopefully I'm not missing a piece of the puzzle here.